runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -29,6 +29,7 @@
 #include "j9consts.h"
 #include "omrgcconsts.h"
 #include "AtomicSupport.hpp"
+#include "ObjectMonitor.hpp"
```

Comment:
Would you please the same explicit include to `ObjectAccessBarrier.cpp` and `mgcalloc.cpp`?

Reply:
I can do this. Is there a particular order the includes should be in? The existing includes aren't quite alphabetical.

Reply:
Well theoretically  includes should be sorted in alphabetical order (in two groups for `*.h` and `*.hpp` files). However in practice it is not used everywhere. `mgcalloc.cpp` has includes sorted so you can include it in proper place. `ObjectAccessBarrier.cpp` has includes not sorted so you can put it somewhere

Reply:
From trying to do this I found a problem. It looks like putting `getInitialLockword` inside `ObjectMonitor.hpp` causes a circular reference when the `ObjectMonitor.hpp` file is included in `ObjectAllocationAPI.hpp`. `ObjectAllocationAPI.hpp` uses `getInitialLockword`.

Include loop:
ObjectMonitor.hpp -> VMHelpers.hpp -> ObjectAllocationAPI.hpp -> ObjectMonitor.hpp

So I think I'll need to move `getInitialLockword` somewhere else or break the circle in a different way.

Reply:
I talked to @amicic and it looks like the include loop can be broken by removing the `VMHelpers.hpp` include inside `ObjectMonitor.hpp`. For what we could tell it wasn't needed. Builds on x86, Power and z were still successful even after removing the include.